---
title: "Analyses report"
author: "Ashfaq Ali"
format: html
editor: visual
---

## Bioinformatic exercise

The present exercise aims to give you the opportunity to demonstrate
your capabilities in analysis of RNASeq data. If you do not have
computational resources to perform the full analysis, you can scale down
the data sets as you see appropriate. You can use any tools/programming
languages you desire, but please provide us with the code you used.
Accessions SRR5223500, SRR5223522, SRR5223543 contain RNASeq data of
non-treated cells while SRR5223505, SRR5223547 and SRR5223570 have been
treated with a stimulatory agent. We wish to determine the effect of the
stimulatory agent on the transcriptome of these cells in order to
understand its effects on cell differentiation.

Questions:

1.  Perform quality control of the sequence data. Are all samples
    appropriate to use?
2.  Map the reads to a reference, perform QC and calculate gene counts.
    Are all samples appropriate to use? -
3.  What is the expression of gene FOXP3 in stimulated vs non-treated
    cells?
4.  What genes are most differentially expressed between treated and
    non-treated cells? -
5.  What is translatability and how would you assess that in the context
    of RNAseq data? Prepare a presentation of your analysis methods and
    answers to the questions above to be presented at the day of
    interview.

> The analyses of the data and brief results are desibed in this file
> which will be further elaborated in the presentation.

## 0. Data download processing analyses

-   Data was downloaded as described in the
    [Analyses_steps.rmd](Analyses_steps.rmd).
-   Data was processed using [bash script](pre_process/hisat.sh).
-   Further QC, statistical analyses, visualizations and functional
    assessment was carried out using the code described in the
    [Analyses_code.rmd](Analyses_code.Rmd) file.

## 1. Perform quality control of the sequence data.

### QC on sequencing data and alignment

To asses the quality of the read data, FASTQC followed by MultiQC was
used. Overall, the data has been sequenced at high depth where most of
the samples have between 34-50 million reads as indicated in the [QC on
reads report](./pre_process/qc/multiqc_report.html). Even though one
sample **SRR5223570** has lower sequencing depth of 8.8 million reads,
the depth is sufficient to reliably quantity expression of most of the
expressed genes.

### Alignment and QC of the alignment

The reads were aligned to human references genome using
[Hisat2](http://daehwankimlab.github.io/hisat2/manual/) and read counts
was performed using
[featureCounts](https://academic.oup.com/bioinformatics/article/30/7/923/232889)
algorithms.

The summary of read alignment and QC is reported in [mapping
qc](./pre_process/mapping_qc/multiqc_report.html) where approximately
**93%** or better alignment rate was observed. The read count statistics
indicated **62%** or better assignment of reads to genes wihich is
reported in the [counts qc
file](./pre_process/counts_qc/multiqc_report.html).

## 2. QC on count data

Feature level QC was performed on the resulting data from alignment and
feature counts in the [directory](./pre_process/results/).

### Violin plots to assess data

![Expressed genes count distribution](results/expressed_genes.png)


## 3. Expression of FOXP3

A higher expression of **FOXP3** was observed in the treated samples vs.
untreated samples.

![FOXP3](results/FOXP3_exp.png)

## 4. Statistical analyses Visualization

The results of differential gene expression analyses using DESeq2 are
reported [here](./results/DE_Results_data.xlsx).

### Volcano plot

![Volcano plot showing differentially expressed
genes](./results/VolcanoPlot_treated_vs_nontreated.png)

### MA plot

![MA plot](./results/MA_treated_vs_untreated.png)

### Heatmap

![Heatmap](./results/HeatMap_treated_vs_Untreated_p_01_lfc_2.png)

## 5. Functional assessment {.tabset}

The results of Gene Set Enrichment analyses bases on various gene sets
available through [msigdb](https://www.gsea-msigdb.org/gsea/index.jsp)
are reported in [this
file](./results/treated_vs_untreated_genesets.xlsx)

### Hallmark

![top 10 hallmark](results/GSEA_Hallmark.png)

### curated_geneset

![curated](results/GSEA_curated_genesets.png)

### computational

![computational](results/GSEA_computational.png)

### GO

![GO](results/GSEA_GO.png) ###Immunological

![Immunological](results/GSEA_Immunological.png)

### Regulation

![Regulation](results/GSEA_Regulation.png)

End
