---
title: "Data_analyses"
author: "Ashfaq Ali"
date: "9/25/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Recquired packages and SetUp

```{r, eval=FALSE, warning=FALSE, message=FALSE}
renv::install("bioc::Biobase")
renv::install("bioc::tximport")
renv::install("bioc::GenomicFeatures")
renv::install("bioc::DESeq2")
renv::install("tidyverse")
renv::install("reshape2")
renv::install("bioc::org.Hs.eg.db")

#renv::install("tidyverse") created problems with ethe select() fucntion.

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(reshape2)
library(DESeq2)
library(org.Hs.eg.db)
```

### Load data

```{r, eval=FALSE, echo=FALSE}

rnaSeq_cts <- read_tsv("./pre_process/results/allSamples.hisat.featureCounts", skip = 1) |> as.data.frame()
rownames(rnaSeq_cts) <- rnaSeq_cts$Geneid

## Clean column names
##
c_names <- colnames(rnaSeq_cts) |>
  str_remove_all(pattern = "/proj/snic2021-23-431/nobackup/hisat_results/") |>
  str_remove_all(pattern = "_grch38.v37.sorted.bam")

colnames(rnaSeq_cts) <- c_names

## Prepare Phenotype data
phenotypes <- data.frame(samples = c("SRR5223500", "SRR5223522", "SRR5223543", "SRR5223505", "SRR5223547", "SRR5223570"), group = rep(c("untreated", "treated"), each = 3))
rownames(phenotypes) <- phenotypes$samples

se <- SummarizedExperiment(assays = list(counts = rnaSeq_cts[, rownames(phenotypes)]), colData = phenotypes, rowData = rnaSeq_cts[, 1:6])
assays(se)$counts <- as.matrix(assays(se)$counts)

se$group <- as.factor(se$group)
se$group <- relevel(se$group, ref = "untreated")
```


## Extract row annotations for later use

```{r, eval=FALSE, echo=FALSE}
annotations <- select(org.Hs.eg.db,
  keys = rnaSeq_cts$Geneid,
  columns = c("SYMBOL", "GENENAME", "ENTREZID"),
  keytype = "ENSEMBL", multiVals = first
)

```


## Gene level QC 

```{r}
rowData(se)$exp_zeors <- rowSums(assays(se)$counts == 0) 
rowData(se)$exp_5 <- rowSums(assays(se)$counts >= 5) 

## remove data with all zeros
se <- se[rowData(se)$exp_zeors !=6, ]
se_f5 <- se[rowData(se)$exp_5 >= 3, ] # Select if gene is expressed in at least 3 samples with 5 or more genes

saveRDS(se , "./pre_process/data_se.rds")
saveRDS(se_f5 , "./pre_process/data_se_reads5.rds")
```


```{r}
se <- readRDS("./pre_process/data_se.rds")
se_f5 <- readRDS("./pre_process/data_se_reads5.rds")
long_data <- melt((assays(se)$counts))
long_data2 <- inner_join(long_data, phenotypes, by = c("Var2" = "samples"))
long_data2$exp_5 <- as.numeric(long_data2$value > 5)

long_data2_f5 <- melt((assays(se_f5)$counts))
long_data2_f5 <- inner_join(long_data2_f5, phenotypes, by = c("Var2" = "samples"))
```



```{r , eval = FALSE, echo = FALSE, fig.width=12, fig.height=15, warning = FALSE, message= FALSE}
p <- ggplot(long_data, aes(x = Var2, y = log2(value))) +
  geom_violin() +
  stat_summary(geom = "point", fun = median) +
  geom_jitter(shape=16,alpha = 0.01) +
  labs(title = "A) Sample wise Read counts", x = "Sample", y = expression("Log"[2] ~ "read counts")) +
  # p + labs(title = "New plot title", subtitle = "A subtitle", tag = "A") + labs(title = NULL)
  labs(caption = "(based on count data after removing genes with zero read count)") +
  theme(axis.line = element_line(colour = "darkblue", size = 1, linetype = 1), axis.text.x = element_text(angle = 90)) +
  # scale_x_discrete(labels=se$sample_id) +
  theme(axis.text.x = element_text(angle = 90))  
    #geom_jitter(, position=position_jitter(0.2), )
p
ggsave(p, filename = "./results/expressed_genes.png", width = 10, height = 8, dpi = 300)


```


There seem to be bi-modal distribution of expressed genes possibly due to different cell types present in the samples.

## "FOXP3"  Expression "ENSG00000049768" is "FOXP3"


```{r , eval = FALSE, echo = FALSE, fig.width=12, fig.height=15, warning = FALSE, message= FALSE}
p2 <- ggplot(long_data2_f5, aes(x = Var2, y = log2(value))) +
  geom_violin() +
  stat_summary(geom = "point", fun = median) +
  geom_jitter(alpha = 0.01) +
  labs(title = "A) Sample wise Read counts", x = "Sample", y = expression("Log"[2] ~ "read counts")) +
  # p + labs(title = "New plot title", subtitle = "A subtitle", tag = "A") + labs(title = NULL)
  labs(caption = "(based on count data after filtering)") +
  theme(axis.line = element_line(colour = "darkblue", size = 1, linetype = 1), axis.text.x = element_text(angle = 90)) +
  # scale_x_discrete(labels=se$sample_id) +
  theme(axis.text.x = element_text(angle = 90))
p2
ggsave(p2, filename = "./results/expressed_genes_f5.png", width = 10, height = 8, dpi = 300)
# ggsave(p, filename = "Results/genes_exrssd_5reads.png", width = 10, height = 8, dpi = 300)
#+ geom_jitter(shape=16, position=position_jitter(0.2))
# png("/Users/cob-aaf/Documents/Projects/PP_4730/NPD_updated/reports/Figures/violin_rawcounts.png", width = 3024, height = 2000, res = 300)
```



## DSeq Object

```{r}


dds <- DESeqDataSet(se_f5, design = ~group)


dds$group <- relevel(dds$group, ref = "untreated")
dds <- estimateSizeFactors(dds)
dds <- rlog(dds)
dds <- DESeq(dds, parallel = TRUE) # Run DESeq

```

```{r}
normalized_counts <- counts(dds, normalized=TRUE)
normalized_counts_long <- melt(normalized_counts)
```


```{r , eval = FALSE, echo = FALSE, fig.width=12, fig.height=15, warning = FALSE, message= FALSE}
p3 <- ggplot(normalized_counts_long, aes(x = Var2, y = log2(value))) +
  geom_violin() +
  stat_summary(geom = "point", fun = median) +
  geom_jitter(alpha = 0.01) +
  labs(title = "A) Sample wise Normalised Read counts", x = "Sample", y = expression("Log"[2] ~ "read counts")) +
  # p + labs(title = "New plot title", subtitle = "A subtitle", tag = "A") + labs(title = NULL)
  labs(caption = "(based on count data after sizeFactor normalization)") +
  theme(axis.line = element_line(colour = "darkblue", size = 1, linetype = 1), axis.text.x = element_text(angle = 90)) +
  # scale_x_discrete(labels=se$sample_id) +
  theme(axis.text.x = element_text(angle = 90))
p3
ggsave(p2, filename = "./results/expressed_genes_f5_normalized.png", width = 10, height = 8, dpi = 300)

```


## FOX P3 expression

```{r}
FOXP3 <- plotCounts(dds , "ENSG00000049768", intgroup = "group", returnData = TRUE)
FOXP3_p <- ggplot(FOXP3, aes(group,count)) + geom_violin() + geom_point()
ggsave(FOXP3_p, filename = "./results/FOXP3_exp.png", dpi = 300)
```


## Results

```{r, eval=FALSE}
treated_vs_untreated <- results(dds, contrast = c("group", "treated", "untreated"), alpha = 0.01) %>%
  as.data.frame() %>%
  rownames_to_column(., var = "gene_id") %>%
  arrange(padj)

treated_vs_untreated <- treated_vs_untreated %>% mutate(sig = padj < 0.01)

treated_vs_untreated <- left_join(treated_vs_untreated, annotations, by = c("gene_id" = "ENSEMBL"))

treated_vs_untreated %>% filter(SYMBOL == "FOXP3")

```



```{r, eval=FALSE}
png("./results/MA_treated_vs_untreated.png", units = "cm", height = 12, width = 15, res = 1500)
DESeq2::plotMA(dds, alpha = 0.2)
dev.off()

```


## Volcano plot

```{r, eval=FALSE}
library(EnhancedVolcano)
ev <- EnhancedVolcano(treated_vs_untreated,
  lab = treated_vs_untreated$gene_id,
  x = "log2FoldChange",
  y = "padj",
  pCutoff = 0.01,
  labSize = 2,
  # ylim = c(0, max(-log10(low_vs_high$padj), na.rm = TRUE) + 1),
  FCcutoff = 2,
  # xlim = c(-2.51,2.51) ,
  #  legendLabels = c("Not significant", "", "adj. P val < 0.05", "adj Pval < 0.05 & effect size >/< 0.01"),
  # transcriptPointSize = 1.5,
  title = "treated vs. untreated",
  subtitle = "",
  boxedLabels = FALSE,
  shadeBins = 4,
  legendLabSize = 10
)
ggsave(plot = ev, filename = "./results/VolcanoPlot_treated_vs_nontreated.png", dpi = 300)
```

## Heatmap

```{r}
library(pheatmap)
vstdds <- vst(dds)
sig_gene <- treated_vs_untreated %>% filter(padj < 0.01 & abs(log2FoldChange) > 2)
mat <- assay(vstdds)[sig_gene$gene_id, ]
mat <- mat - rowMeans(mat)
anno <- as.data.frame(colData(vstdds)[, c("samples", "group")])

png("./results/HeatMap_treated_vs_Untreated_p_01_lfc_2.png", units = "cm", height = 35, width = 15, res = 300)
pheatmap(mat, annotation_col = anno, fontsize_row = 6)
dev.off()

```



```{r}
writexl::write_xlsx(treated_vs_untreated, "./results/DE_Results_data.xlsx")
```

## Gene set enrichment analyses using MSigDb

### Load gene sets

```{r eval=FALSE, include=FALSE, message=FALSE, warning=FALSE}
library(msigdbr)

m_df <- msigdbr(species = "Homo sapiens")
hm_gsig <- m_df %>%
  filter(gs_cat == "H") %>%
  dplyr::select(gs_name, gene_symbol) # Hallmark genesets
C5_gsig <- m_df %>%
  filter(gs_cat == "C5") %>%
  dplyr::select(gs_name, gene_symbol) # Go terms
C2_gsig <- m_df %>%
  filter(gs_cat == "C2") %>%
  dplyr::select(gs_name, gene_symbol) # Curated gene sets (Kegg pathways)
C3_gsig <- m_df %>%
  filter(gs_cat == "C3") %>%
  dplyr::select(gs_name, gene_symbol) # Regulatory
C4_gsig <- m_df %>%
  filter(gs_cat == "C4") %>%
  dplyr::select(gs_name, gene_symbol) # computational
C7_gsig <- m_df %>%
  filter(gs_cat == "C7") %>%
  dplyr::select(gs_name, gene_symbol) # C7 (immunologic signatures, 4872 gene sets)
C6_gsig <- m_df %>%
  filter(gs_cat == "C6") %>%
  dplyr::select(gs_name, gene_symbol) # Oncogenic
C8_gsig <- m_df %>%
  filter(gs_cat == "C8") %>%
  dplyr::select(gs_name, gene_symbol) # Cell types

```

### Gene Set Enrichment analyses

```{r eval=FALSE, include=FALSE}
set.seed(1234)
library(clusterProfiler)
treated_vs_untreated_genesets <- list()
W1 <- treated_vs_untreated$log2FoldChange
names(W1) <- treated_vs_untreated$SYMBOL
W1 <- sort((W1), decreasing = TRUE)
W1 <- W1[!duplicated(names(W1))]

W2 <- treated_vs_untreated$baseMean
names(W2) <- treated_vs_untreated$SYMBOL
W2 <- sort((W2), decreasing = TRUE)
W2 <- W2[!duplicated(names(W2))]
# mutate(rank = rank(log2FoldChange,  ties.method = "random"))
treated_vs_untreated_genesets$Hallmark <- GSEA(W1,
  TERM2GENE = hm_gsig, exponent = 1,
  # nPerm = 10000,
  pvalueCutoff = 0.2, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$curated_genesets <- GSEA(W1, TERM2GENE = C2_gsig, exponent = 1, pvalueCutoff = 0.05, minGSSize = 3, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$computational <- GSEA(W1,
  TERM2GENE = C3_gsig, exponent = 1,
  # nPerm = 10000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$GO <- GSEA(W1,
  TERM2GENE = C5_gsig, exponent = 1,
  # nPerm = 1000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$Immunological <- GSEA(W1,
  TERM2GENE = C7_gsig, exponent = 1,
  # nPerm = 1000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$Regulation <- GSEA(W1,
  TERM2GENE = C3_gsig, exponent = 1,
  # nPerm = 1000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())
treated_vs_untreated_genesets$cell_types <- GSEA(W1,
  TERM2GENE = C8_gsig, exponent = 1,
  # nPerm = 1000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

treated_vs_untreated_genesets$cell_types_exp_rank <- GSEA(W2,
  TERM2GENE = C8_gsig, exponent = 1,
  # nPerm = 1000,
  pvalueCutoff = 0.05, minGSSize = 5, maxGSSize = 200, verbose = FALSE, by = "fgsea", eps = 0
) %>%
  as.data.frame() %>%
  tibble() %>%
  mutate(enrichment = ifelse(NES > 0, "Up-regulated", "Down-regulated"), Description = stringr::str_replace_all(Description, pattern = "_", replacement = " ") %>% str_to_sentence())

#
writexl::write_xlsx(treated_vs_untreated_genesets, "./results/treated_vs_untreated_genesets.xlsx")

```


## Select top 10

```{r}
top_10 <- list()
for (i in 1:length(names(treated_vs_untreated_genesets))) {
  top_10[[i]] <- treated_vs_untreated_genesets[[i]] |>
    group_by(enrichment) |>
    top_n(10, NES) |>
    ungroup()
}
names(top_10) <- names(treated_vs_untreated_genesets)
```


```{r}
knitr::kable(top_10) %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive", "bordered"))
```

### Plot GSEA results

```{r  eval=FALSE, include=FALSE, message=FALSE, warning=FALSE, include=FALSE}
for (i in 1:length(top_10)) {
  name <- names(top_10)[i]
  g <- list()
  g[[i]] <- ggplot(top_15[[i]], aes(reorder(Description, NES), NES)) +
    geom_segment(aes(reorder(Description, NES), xend = Description, y = 0, yend = NES)) +
    geom_point(
      size = 5, aes(fill = enrichment),
      shape = 21, stroke = 2
    ) +
    scale_fill_manual(values = c(
      "Down-regulated" = "dodgerblue",
      "Up-regulated" = "firebrick"
    )) +
    coord_flip() +
    labs(
      x = "Gene Set", y = "Normalized Enrichment Score",
      title = paste(name)
    ) +
    # xlab(str_to_sentence(treated_vs_untreated_genesets[[i]]$Description)) +
    theme_minimal()
  png(file.path("./results/", paste0("GSEA_", paste(name), ".png")), width = 1600, height = 1800, units = "px", res = 180)
  print(g[[i]])
  dev.off()
}
```

